pkg/cmd/roachtest/decommission.go

```diff
@@ -374,318 +451,486 @@ func runDecommissionAcceptance(ctx context.Context, t *test, c *cluster) {
 		return res
 	}
 
-	t.l.Printf("decommissioning first node from the second, polling the status manually\n")
+	// expectColumn constructs a matching regex for a given column (identified
+	// by its column index).
+	expectColumn := func(column int, columnRegex []string, numRows, numCols int) [][]string {
+		var res [][]string
+		for r := 0; r < numRows; r++ {
+			build := []string{}
+			for c := 0; c < numCols; c++ {
+				if c == column {
+					build = append(build, columnRegex[r])
+				} else {
+					build = append(build, `.*`)
+				}
+			}
+			res = append(res, build)
+		}
+		return res
+	}
+	// expectCell constructs a matching regex for a given cell (identified by
+	// its row and column indexes).
+	expectCell := func(row, column int, regex string, numRows, numCols int) [][]string {
+		var res [][]string
+		for r := 0; r < numRows; r++ {
+			build := []string{}
+			for c := 0; c < numCols; c++ {
+				if r == row && c == column {
+					build = append(build, regex)
+				} else {
+					build = append(build, `.*`)
+				}
+			}
+			res = append(res, build)
+		}
+		return res
+	}
+
 	retryOpts := retry.Options{
 		InitialBackoff: time.Second,
 		MaxBackoff:     5 * time.Second,
-		Multiplier:     1,
+		Multiplier:     2,
 	}
-	if err := retry.WithMaxAttempts(ctx, retryOpts, 50, func() error {
-		o, err := decommission(ctx, 2, c.Node(1),
-			"decommission", "--wait=none", "--format=csv")
+
+	// Phew, now for the actual test..
+
+	// Partially decommission then recommission a random node, from another
+	// random node. Run a couple of status checks while doing so.
+	{
+		targetNode, runNode := getRandNode(), getRandNode()
+		t.l.Printf("partially decommissioning n%d from n%d\n", targetNode, runNode)
+		o, err := decommission(ctx, c.Node(targetNode), runNode,
+			"--wait=none", "--format=csv")
 		if err != nil {
 			t.Fatalf("decommission failed: %v", err)
 		}
 
 		exp := [][]string{
 			decommissionHeader,
-			{"1", "true", "0", "true", "false"},
-			decommissionFooter,
+			{strconv.Itoa(targetNode), "true", `\d+`, "true", "decommissioning", "false"},
+		}
+		if err := matchCSV(o, exp); err != nil {
+			t.Fatal(err)
 		}
 
-		return matchCSV(o, exp)
-	}); err != nil {
-		t.Fatal(err)
-	}
+		// Check that `node status` reflects an ongoing decommissioning status
+		// for the second node.
```

Comment:
This won't flake, right? The decommission status is propagates async, but behind the scenes we always hit KV here?

Reply:
Yup.